### DISCUSSION TITLE
How much should we devs trust Rope's unit tests?

### DISCUSSION THREAD

**Discussion by edreamleo** [python-rope/rope#676] [General]
In the past several weeks I have proposed several PRs that changed Rope's organization. Failing unit tests guided my work :-)

My underlying assumption was that all would be well when all tests pass. My experience leads me to trust that assumption, but I don't have any *proof* that my assumption is valid.

@lieryan I would like your opinion, please.

**Comment by lieryan** [comment]
Rope's unittest is generally quite good. At least all the most commonly used core functionalities are generally quite reasonably thoroughly well tested.

There are a few places where the test are lacking, the DOA and fscommands comes to mind, but they're kinda fringe.
